index.bs

```diff
@@ -488,13 +490,238 @@ partial interface NeuralNetworkContext {
+        let cellInput = nn.squeeze(nn.slice(input, [slice, 0, 0], [slice + 1, -1, -1]), [0]);
+
+        let result = nn.gruCell(
+          hiddenSize, cellInput, cellWeight[slot], cellRecurrentWeight[slot],
+          cellHidden[slot], cellBias[slot], cellRecurrentBias[slot], 
+          resetAfter, layout, activations);
+
+        cellOutput = (cellOutput ? nn.concat([cellOutput, result], 0) : result);
+      }
+
+      hidden = cellOutput;
+
+      if (returnSequence) {
+        cellOutput = nn.reshape(cellOutput, [1, slots, -1, hiddenSize]);
+        sequence = (sequence ? nn.concat([sequence, cellOutput], 0) : cellOutput);
+      }
+    }
+
+    return (sequence ? [hidden, sequence] : [hidden]);
+    </pre>
+    </div>
+</div>
+
+### gruCell ### {#api-neuralnetworkcontext-grucell}
+A single time step of the Gated Recurrent Unit (GRU) recurrent network using an update gate and a reset gate to compute the hidden state that rolls into the output across the temporal sequence of a recurrent network.
+<script type=idl>
+partial interface NeuralNetworkContext {
+  Operand gruCell(long hiddenSize, Operand input, Operand weight, 
+                  Operand recurrentWeight, Operand hidden,
+                  optional Operand bias, optional Operand recurrentBias,
+                  optional boolean resetAfter = true,
+                  optional RecurrentNetworkWeightLayout layout = "zrn",
+                  optional sequence<RecurrentNetworkActivation> activations);
+};
+</script>
+<div algorithm=grucell>
+    **Arguments:**
+        - *hiddenSize*: a {{long}} scalar. The value of the second dimension of the output tensor shape. It indicates the number of features in the hidden state.
+        - *input*: an {{Operand}}. The input 2-D tensor of shape [batch_size, input_size]. 
+        - *weight*: an {{Operand}}. The 2-D input weight tensor of shape [3 * hidden_size, input_size]. The ordering of the weight vectors in the first dimension of the tensor shape is specified according to the *layout* argument.
+        - *recurrentWeight*: an {{Operand}}. The 2-D recurrent weight tensor of shape [3 * hidden_size, hidden_size]. The ordering of the weight vectors in the first dimension of the tensor shape is specified according to the *layout* argument.
+        - *hidden*: an {{Operand}}. The 2-D input hidden state tensor of shape [batch_size, hidden_size].
+        - *bias*: an optional {{Operand}}. The 1-D input bias tensor of shape [3 * hidden_size]. The ordering of the bias vectors in the first dimension of the tensor shape is specified according to the *layout* argument.
+        - *recurrentBias*: an optional {{Operand}}. The 1-D recurrent bias tensor of shape [3 * hidden_size]. The ordering of the bias vectors in the first dimension of the tensor shape is specified according to the *layout* argument.
+        - *resetAfter*: an optional {{boolean}} indicating whether to apply the reset gate after or before matrix multilication. Default to true.
```

Comment:
`resetAfterLinear` that aligns to [ONNX GRU attribute - `linear_before_reset`](https://github.com/onnx/onnx/blob/master/docs/Operators.md#GRU)?

Reply:
Thanks for the explanation and reference. It sounds good.